src/views/home/components/list/list.view.tsx

```diff
@@ -0,0 +1,79 @@
+import { FC, useState } from "react";
+
+import useListStyles from "src/views/home/components/list/list.styles";
+import Card from "src/views/shared/card/card.view";
+import { ReactComponent as SearchIcon } from "src/assets/icons/search.svg";
+import Typography from "src/views/shared/typography/typography.view";
+import TokenIcon from "src/views/shared/token-icon/token-icon.view";
+import { Chain, Token } from "src/domain";
+
+interface ListProps {
+  placeholder: string;
+  list:
+    | { type: "chain"; items: Chain[]; onClick: (chain: Chain) => void }
+    | { type: "token"; items: Token[]; onClick: (token: Token) => void };
+}
+
+const List: FC<ListProps> = ({ placeholder, list }) => {
+  const classes = useListStyles();
+  const [values, setValues] = useState(list);
+
+  const filterChain = (chain: Chain, value: string) =>
+    chain.name.toUpperCase().includes(value.toUpperCase());
+  const filterToken = (token: Token, value: string) =>
+    token.name.toUpperCase().includes(value.toUpperCase()) ||
+    token.symbol.toUpperCase().includes(value.toUpperCase());
```

Comment:
I'd move these filtering functions out of the component and make them take care of the whole filtering job, i.e they receive a `List` and a `value` and return the filtered `List`.

> Notice that I removed `type: "chain"` and `type: "token"` in the returned list since we have these from the spread.

```typescript
const filterChainList = (chainList: ChainList, value: string): ChainList => ({
  ...chainList,
  items: chainList.items.filter((chain) =>
    chain.name.toUpperCase().includes(value.toUpperCase())
  ),
});

const filterTokenList = (tokenList: TokenList, value: string): TokenList => ({
  ...tokenList,
  items: tokenList.items.filter(
    (token) =>
      token.name.toUpperCase().includes(value.toUpperCase()) ||
      token.symbol.toUpperCase().includes(value.toUpperCase())
  ),
});
```

By encapsulating the filtering in the functions we simplify the `onChange` handler:

> Notice also here the `trim()` to remove spaces from user input

```typescript
const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
  const value = event.target.value.trim();
  const filtered: List =
    list.type === "chain" ? filterChainList(list, value) : filterTokenList(list, value);
  setValues(filtered);
};
```